webdriver/tests/take_element_screenshot/user_prompts.py

```diff
@@ -15,56 +15,100 @@ def take_element_screenshot(session, element_id):
         )
     )
 
+# In some cases, such as "dismiss and notify" & "accept and notify" dismiss
+# the dialog and throw an error. This fixture checks those cases.
+@pytest.fixture
+def check_user_prompt_closed_with_exception(session, create_dialog):
+    def check_user_prompt_closed_with_exception(dialog_type):
+        session.url = inline("<input/>")
+        element = session.find.css("input", all=False)
+
+        create_dialog(dialog_type, text=dialog_type)
+        assert session.alert.text == dialog_type
+
+        response = take_element_screenshot(session, element.id)
+        assert_error(response, "unexpected alert open")
+
+        # Since dismiss and notify or accept and notify would have dismissed,
+        # retrying the screenshot will succeed
+        response = take_element_screenshot(session, element.id)
+        value = assert_success(response)
+
+        assert_png(value)
```

Comment:
Is there a less expensive command we can run to test that it worked successfully?

Reply:
Looking into assert_png, it already does a bare minimum amount of work:

- Decode Base64
- Get and check mime_type.

I don't think there is a direct way to reduce the work here.

I thought about possibly just omitting the assert_png entirely from user_prompts.py files. After all, we are validating the user_prompt behavior around screenshots, not the screenshot itself.

However, I think there is value in keeping them there in case of a browser implementation doing something such as returning success but not successfully taking a screenshot.